apps/astarte_data_updater_plant/lib/astarte_data_updater_plant/data_updater/impl.ex

```diff
@@ -1188,6 +1200,28 @@ defmodule Astarte.DataUpdaterPlant.DataUpdater.Impl do
     {:ok, new_state}
   end
 
+  @impl true
+  def handle_continue({:malformed_capabilities_message, payload, timestamp}, state) do
+    :telemetry.execute(
+      [:astarte, :data_updater_plant, :data_updater, :discarded_capabilities_message],
+      %{},
+      %{realm: state.realm}
+    )
+
+    error_metadata = %{
+      "payload" => payload
```

Comment:
The payload is a binary that might be long and truncated. Let's `base64` it, as done e.g. for unexpected internal messages
https://github.com/astarte-platform/astarte/blob/0e32393d972a8c545fc53d3efc9df2f762997f3f/apps/astarte_data_updater_plant/lib/astarte_data_updater_plant/data_updater/impl.ex#L970-L993